### PR TITLE
Create tag on ui

### PR DESCRIPTION
**Pull request by a1012112796**
1. Support create single tag directly 
Peek 2020-11-09 00-11

2. support create tag with message from create release ui
Peek 2020-11-09 12-25

### CODE REVIEW

**Comment by lafriks** [comment]
Wouldn't it be better to to that from specific form like release where you can also input tag message?

**Comment by silverwind** [comment]
Agree, tags have messages, even if we don't display them yet, so there needs to be a form probably (thought I personally don't mind creating tags on CLI only).

**Comment by 6543** [comment]
I would considder this a feature or enhancement but not a bugfix

**Comment by a1012112796** [comment]
@codecov-io waiting result :)

File: modules/git/repo_tag.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func (repo *Repository) CreateTag(name, revision string) error {
 
 // CreateAnnotatedTag create one annotated tag in the repository
 func (repo *Repository) CreateAnnotatedTag(name, message, revision string) error {
-	_, err := NewCommand("tag", "-a", "-m", message, "--", name, revision).RunInDir(repo.Path)
+	_, err := NewCommand("tag", "-a", name, revision, "-m", message).RunInDir(repo.Path)
```
Comment (zeripath) [inline modules/git/repo_tag.go]: Why the change in the order of arguments here?

From the git manual 

```
NAME
       git-tag - Create, list, delete or verify a tag object signed with GPG

SYNOPSIS
       git tag [-a | -s | -u <keyid>] [-f] [-m <msg> | -F <file>] [-e]
               <tagname> [<commit> | <object>]
       git tag -d <tagname>...
       git tag [-n[<num>]] -l [--contains <commit>] [--no-contains <commit>]
               [--points-at <object>] [--column[=<options>] | --no-column]
               [--create-reflog] [--sort=<key>] [--format=<format>]
               [--[no-]merged [<commit>]] [<pattern>...]
       git tag -v [--format=<format>] <tagname>...
```

You need to rearrange these arguments:

```suggestion
	_, err := NewCommand("tag", "-a", "-m", message, name, revision).RunInDir(repo.Path)
```

However, there really should be a way of terminating the options, usually that is through `--`. Does that really not work here? if it doesn't we need to ensure that both name and revision do not start with an initial `-`.

Comment (zeripath) [reply modules/git/repo_tag.go]: Otherwise you can cause a hang simply by passing in a name or revision equal to `-e`

Comment (a1012112796) [reply modules/git/repo_tag.go]: Sorry, I forget it, It's not an necessary change. I have change it to previous version now.

File: routers/repo/branch.go [hunk 2]
```diff
@@ -383,7 +384,16 @@ func CreateBranch(ctx *context.Context) {
...
 	var err error
-	if ctx.Repo.IsViewBranch {
+
+	if form.CreateTag {
+		if ctx.Repo.IsViewBranch {
+			err = release_service.CreateNewTag(ctx.User, ctx.Repo.Repository, ctx.Repo.BranchName, form.NewBranchName, "")
+		} else if ctx.Repo.IsViewTag {
+			err = release_service.CreateNewTag(ctx.User, ctx.Repo.Repository, ctx.Repo.CommitID, form.NewBranchName, "")
+		} else {
+			err = release_service.CreateNewTag(ctx.User, ctx.Repo.Repository, ctx.Repo.BranchName, form.NewBranchName, "")
```
Comment (6543) [inline routers/repo/branch.go:392]: this if can be simplified

File: options/locale/locale_en-US.ini [hunk 3]
```diff
@@ -1892,6 +1892,8 @@ release.tag_name_invalid = The tag name is not valid.
 release.tag_already_exist = This tag name already exists.
 release.downloads = Downloads
 release.download_count = Downloads: %s
+release.add_tag_msg = use the title and content of release as tag message.
```
Comment (6543) [inline options/locale/locale_en-US.ini]:
```suggestion
release.add_tag_msg = Use the title and content of release as tag message.
```

File: options/locale/locale_en-US.ini [hunk 4]
```diff
@@ -1892,6 +1892,8 @@ release.tag_name_invalid = The tag name is not valid.
...
+release.add_tag = create tag only
```
Comment (6543) [inline options/locale/locale_en-US.ini]:
```suggestion
release.add_tag = Create Tag Only
```

**Review by 6543 — APPROVED** [review]

**Review by zeripath — APPROVED** [review]
